src/components/selectorModal.component.ts

```diff
@@ -12,15 +12,44 @@ export interface Item {
 })
 export class SelectorModalComponent {
     @Input() items: Item[]
+    shellFilter = '';
+    filteredItems: Item[]
+    selectedOption = 0
 
     constructor (
         public modalInstance: NgbActiveModal,
     ) { }
 
+    ngOnInit() {
+        this.filteredItems = this.items
+    }
+
+    onSearch() {
+        this.filteredItems = Object.assign([], this.items).filter(
+            item => item.name.toLowerCase().indexOf(this.shellFilter.toLowerCase()) > -1
+        )
+        this.selectedOption = 0
+    }
+
     @HostListener(`document:keyup`, ['$event']) onKey ($event: KeyboardEvent) {
         if ($event.key >= '0' && $event.key <= '9') {
             let index = $event.key.charCodeAt(0) - '0'.charCodeAt(0)
-            this.modalInstance.close(this.items[index])
+            if (index < this.filteredItems.length) {
```

Comment:
Never happens since the list is always empty after typing a digit in to the filter box